src/sinks/influxdb/mod.rs

```diff
@@ -392,25 +392,38 @@ pub mod test_util {
 
     pub(crate) async fn onboarding_v1(endpoint: &str) -> String {
         let database = next_database();
-        query_v1(endpoint, &format!("drop database {}", database)).await;
         let status = query_v1(endpoint, &format!("create database {}", database))
             .await
             .status();
-        assert!(
-            status == http::StatusCode::OK,
-            format!("UnexpectedStatus: {}", status)
-        );
+        assert_eq!(status, http::StatusCode::OK, "UnexpectedStatus: {}", status);
+        // Some times InfluxDB will return OK before it can actually
+        // accept writes to the database, leading to test failures. Test
+        // this with empty writes and loop if it reports the database
+        // does not exist yet.
+        loop {
```

Comment:
I think it'd be good to have this loop break after, say, 10 iterations (5 seconds) to avoid this test hanging forever.